flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/lookup/cache/DefaultLookupCache.java

```diff
@@ -97,30 +97,36 @@ public static DefaultLookupCache fromConfig(ReadableConfig config) {
 
     @Override
     public void open(CacheMetricGroup metricGroup) {
-        // Initialize Guava cache
-        CacheBuilder<Object, Object> guavaCacheBuilder = CacheBuilder.newBuilder();
-        if (expireAfterAccessDuration != null) {
-            guavaCacheBuilder.expireAfterAccess(expireAfterAccessDuration);
-        }
-        if (expireAfterWriteDuration != null) {
-            guavaCacheBuilder.expireAfterWrite(expireAfterWriteDuration);
-        }
-        if (maximumSize != null) {
-            guavaCacheBuilder.maximumSize(maximumSize);
-        }
-        if (ticker != null) {
-            guavaCacheBuilder.ticker(ticker);
+        synchronized (this) {
+            // The cache should already been opened if guava cache is not null
+            if (guavaCache != null) {
+                return;
```

Comment:
We may need to register the metrics to the given `metricGroup` again. Every subtask will retrieve the shared cache and register cache metrics to the operator metric group.